packages/millicast-sdk/src/utils/Codecs.js

```diff
@@ -193,10 +193,91 @@ export function extractH26xMetadata (encodedFrame, codec) {
   }
 }
 
+function parseUUID (uuid) {
+  return uuid.replace(/-/g, '')
+    .match(/.{1,2}/g)
+    .map(byte => parseInt(byte, 16))
+}
+
+function createSEIMessageContent (uuid, payload) {
+  const uuidArray = new Uint8Array(parseUUID(uuid))
+  const payloadArray = new TextEncoder().encode(payload)
+
+  const content = new Uint8Array(uuidArray.length + payloadArray.length)
+  content.set(uuidArray)
+  content.set(payloadArray, uuidArray.length)
+
+  return content
+}
+
+function createSEITypeAndSize (content) {
+  const payloadSize = []
+  const ffBytes = Math.floor(content.byteLength / 255)
+  const lastPayloadTypeByte = content.byteLength % 255
+  for (let i = 0; i < ffBytes; i++) {
+    payloadSize.push(0xFF)
+  }
+  payloadSize.push(lastPayloadTypeByte)
+
+  return new Uint8Array([0x05, ...payloadSize])
+}
+
+function createSEIMessageContentWithPrevensionBytes (content) {
+  const preventionByteArray = []
+
+  for (let i = 0; i < content.byteLength; i++) {
+    if (i + 2 < content.byteLength && [0x00, 0x01, 0x02, 0x03].includes(content[i + 2]) && content[i] === 0x00 && content[i + 1] === 0x00) {
+      preventionByteArray.push(content[i])
+      preventionByteArray.push(content[i + 1])
+      i += 2
+      preventionByteArray.push(0x03)
+    } else {
+      preventionByteArray.push(content[i])
+    }
+  }
+
+  // trailing bits
+  preventionByteArray.push(0x80)
+
+  return new Uint8Array(preventionByteArray)
+}
+
+function createSEINalu ({ uuid, payload }, codec) {
+  const startCode = [0x00, 0x00, 0x00, 0x01]
+  const header = [0x66] // 0b01100110
+  const content = createSEIMessageContent(uuid, payload)
+  const seiTypeAndSize = createSEITypeAndSize(content)
+  const contentWithPreventionBytes = createSEIMessageContentWithPrevensionBytes(content)
+
+  const naluWithSEI = new Uint8Array(startCode.length + header.length + seiTypeAndSize.length + contentWithPreventionBytes.length)
+  naluWithSEI.set(startCode)
+  naluWithSEI.set(header, startCode.length)
+  naluWithSEI.set(seiTypeAndSize, startCode.length + header.length)
+  naluWithSEI.set(contentWithPreventionBytes, startCode.length + header.length + seiTypeAndSize.length)
+
+  return naluWithSEI
+}
+
 export function addH26xSEI ({ uuid, payload }, encodedFrame, codec) {
   if (codec !== 'h264' && codec !== 'h265') {
     throw new Error(`Unsupported codec ${codec}`)
   }
-  // TODO: create new Uint8Array as NAL unit. NAL unit format is start code + header + sei type + payload then push to the original NAL units array
-  return encodedFrame
+  if (uuid === '' || payload === '') {
+    return
```

Comment:
We should throw error in this function instead of return which is not good for developer to debug
```suggestion
    throw new Error('uuid and payload cannot be empty')
```